models.rb

```diff
@@ -15,6 +15,16 @@ def validate
     super
     errors.add(:name, 'cannot be empty') if !name || name.empty?
   end
+
+  def remove(client)
```

Comment:
I don't think you need to take client as a parameter here, since this is an instance method that will be called like `client.delete` - the client instance is already there, there is no reason to send it a copy of itself.